### ISSUE TITLE
Dreaded "Addon is corrupt" error

### ISSUE THREAD

**Issue by hydroculator** [onemen/TabMixPlus#274]
I have been unable to install the addon, and receive the "addon is corrupt" message.
I have read and followed every instruction and suggestion I can find in all the threads under Issues and the install instructions multiple times. I've manually verified all the advanced preferences items, I've cleared the startup cache. I've tried every version of config.js I have seen in all the threads. I've tried running the code in the browser console. I don't know what else to try. 
I deleted the cache folder in AppData\Local.  
I've gotten this to work on my personal computer, but this one is managed by IT. I have admin rights on it, but could there be something else that I could check to see if something is overriding this somehow?

Thanks for your help and the great addon.

Profile Folder: 
C:\Users\redacted\AppData\Roaming\Mozilla\Firefox\Profiles\0pot7u9c.default-esr
Scripts are here (I have tried both the extension only and the script and extensions zip file): 
C:\Users\redacted\AppData\Roaming\Mozilla\Firefox\Profiles\0pot7u9c.default-esr\chrome\utils
BootstrapLoader.jsm
chrome.manifest
RDFDataSource.jsm
RDFManifestConverter.jsm
xPref.jsm

Application Binary:
C:\Program Files\Mozilla Firefox\firefox.exe
config.js is here:
C:\Program Files\Mozilla Firefox\config.js
config-prefs.js is here:
C:\Program Files\Mozilla Firefox\defaults\pref\config-prefs.js

xpinstall.signatures.required: locked false
extensions.install_origins.enabled: locked false
general.config.obscure_value: 0
extensions.experiments.enabled: true
general.config.sandbox_enabled: false
devtools.chrome.enabled:  true
general.config.filename: config.js

However, this returns True.
1709586492507	addons.xpi	WARN	Invalid XPI: Error: File C:\downloads\tab_mix_plus-dev-build.xpi does not contain a valid manifest(resource://gre/modules/addons/XPIInstall.jsm:685:11) JS Stack trace: loadManifest@XPIInstall.jsm:685:11

Contents of C:\Program Files\Mozilla Firefox\config.js:

// skip 1st line
lockPref('xpinstall.signatures.required', false);
lockPref('extensions.install_origins.enabled', false);

try {
  const cmanifest = Cc['@mozilla.org/file/directory_service;1'].getService(Ci.nsIProperties).get('UChrm', Ci.nsIFile);
  cmanifest.append('utils');
  cmanifest.append('chrome.manifest');
  Components.manager.QueryInterface(Ci.nsIComponentRegistrar).autoRegister(cmanifest);

  const objRef = ChromeUtils.import('resource://gre/modules/addons/XPIDatabase.jsm');
  objRef.XPIDatabase.SIGNED_TYPES.delete("extension");

  Cu.import('chrome://userchromejs/content/BootstrapLoader.jsm');
} catch (ex) {};

try {
  Cu.import('chrome://userchromejs/content/userChrome.jsm');
} catch (ex) {};

**Comment by onemen** [comment]
https://onemen.github.io/tabmixplus-docs/other/installation/

**Comment by onemen** [comment]
What is yor firefox version?

**Comment by hydroculator** [comment]
123.0.1 (64-bit)

**Comment by ozzyjr** [comment]
I also had difficulties with the new documentation and even repeating all the steps several times, I continued to receive the message that the "addon is corrupt".
I decided to redo the steps, but instead of using the documentation files, I used the backup files from an installation that I used until last week and it worked the first time. I used the "test-build-1.0.0-pre.19.20240309.1.xpi" version and updated the extensions, updating the TMP to the latest version, 1.19.1.

This backup used the "old" method, which used userChromeJS

I'm using Firefox Nightly 125.0a1 (2024-03-11) (64-bit)

I would like to understand why it worked one way and not another.

**Comment by onemen** [comment]
@ozzyjr,

can you provide copy of the script files you where using

**Comment by ozzyjr** [comment]
Of course @onemen 

I'm on a MacOs Sonoma 14.4

Files
[files.zip](https://github.com/onemen/TabMixPlus/files/14562423/files.zip)

**Comment by onemen** [comment]
@ozzyjr,

I will try to check these files and the files from the dcos and see if I can reproduce this issue.

beside the different files is there anything else that i can do to improve the installation documents ?

**Comment by ozzyjr** [comment]
@onemen

No, the installation documents are fine. The only problem was the corrupt addon message that I couldn't resolve using the documentation files.

Other than that, your service is as excellent as ever.

**Comment by onemen** [comment]
@ozzyjr ,

I have just tested the files from installation documents on my VirtualBox MacOS and I was able to install Tab Mix Plus without any problem.

One important different between [utils](https://github.com/onemen/TabMixPlus/files/14075742/utils_extensions_and_scripts.zip) zip file in the documents and the [files](https://github.com/onemen/TabMixPlus/files/14562423/files.zip) you are using is that in you zip files `utils` folder is inside `chrome` folder and the documents `utils` folder is at the root of the zip file.

I just updated the instruction
![utils](https://github.com/onemen/TabMixPlus/assets/3650909/a3a638b1-0c55-4acd-9d62-13dc6096efd6)

**Comment by ozzyjr** [comment]
@onemen 

OK. Thanks for the explanation and help.

**Comment by tekert** [comment]
Same problem on windows, installed everything correctly, like always, but this time the corrupt message shows up.

**Comment by koppor** [comment]
Same issue here on Firefox `125.0.2 (64-bit)`, Windows 10. Did not have TabMixPlus installed before.

Should I change `general.config.obscure_value` from 13 to 0?

**Comment by onemen** [comment]
> Same issue here on Firefox `125.0.2 (64-bit)`, Windows 10. Did not have TabMixPlus installed before.
> 
> Should I change `general.config.obscure_value` from 13 to 0?

we already have `pref("general.config.obscure_value", 0);` in the file `config-prefs.js` - that should be copied to '[PATH TO FIREFOX.EXE]/defaults/pref', if the value for "general.config.obscure_value" is 13 that is indication that you did not install `Firefox Scripts` properly.

**Comment by koppor** [comment]
Thank you so much for the support! Yes, that's it! TerraCopy worked only the second time when copying.

**Comment by Squall-Leonhart** [comment]
I have this occuring here too, just not for TMP, but other extensions such as bbCodeWebEx and enhanced-h264ify,

issue is directly caused with the placing of the config.js in the installer folder as removing it and restarting it allows them to install again.

**Comment by onemen** [comment]
Verify that all your firefox-scripts are up-to-date 
https://github.com/onemen/TabMixPlus/files/14075743/fx-folder.zip
https://github.com/onemen/TabMixPlus/files/14075742/utils_extensions_and_scripts.zip

**Comment by onemen** [comment]
@Squall-Leonhart read #292

**Comment by Albirew** [comment]
for the record, some installations (like mine) have a `defaults/pref/antibeacon.js` file containing `pref("general.config.filename", "mozilla.cfg");` and `pref("general.config.obscure_value", 0);`.
In this case, I added at the end of this file `pref('general.config.sandbox_enabled', false);`, and pasted all content of `config.js` into already existing `mozilla.cfg`
